sql/core/src/main/scala/org/apache/spark/sql/sources/DataSourceStrategy.scala

```diff
@@ -176,6 +176,9 @@ private[sql] object DataSourceStrategy extends Strategy {
       case expressions.Contains(a: Attribute, Literal(v: String, StringType)) =>
         Some(sources.StringContains(a.name, v))
 
+      case p: expressions.Predicate =>
```

Comment:
Ok. I am also not very sure about this modification. But seems there is no other proper approach to check possible error in this place.

I will revert this part first.